File: TypeScript/3.无重复字符的最长子串 Array.ts
function lengthOfLongestSubstring(s: string): number {
    let ans: number = 0
    if (s.length === 0)
        return ans
    const arr = new Array<boolean>(128).fill(false)
    let [left, right] = [0, 0]
    while (right < s.length) {
        if (arr[s.charCodeAt(right)] == false) {
            arr[s.charCodeAt(right)] = true
            ans = Math.max(ans, right - left + 1)
            ++right
        } else {
            arr[s.charCodeAt(left)] = false
            ++left
        }
    }
    return ans
};

File: Golang/3.无重复字符的最长子串 array.go
package leetcode

func lengthOfLongestSubstring(s string) int {
	if len(s) == 0 {
		return 0
	}
	ans := 0
	arr := [128]bool{}
	left, right := 0, 0
	for right < len(s) {
		if arr[s[right]] == false {
			arr[s[right]] = true
			ans = maxInt(ans, right-left+1)
			right++
		} else {
			arr[s[left]] = false
			left++
		}
	}
	return ans
}

func maxInt(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
